### PR TITLE
Upgrade deps for node v14+

### PR DESCRIPTION
**Pull request by devinivy**
Upgrade hoek, teamwork, validate, lab, code, and typescript for node v14+ support.

If this looks good, this will be the last work going into podium v5.

### CODE REVIEW

**Comment by Marsup** [comment]
Apparently, the tests fail because lab doesn't include a new kind of error around [here](https://github.com/hapijs/lab/blob/master/lib/modules/types.js#L27-L44), the new code is [2820](https://github.com/microsoft/TypeScript/blob/bc161c0dd8f44df74247c29f5fa5eb97dbae457b/src/compiler/diagnosticMessages.json#L3418-L3421). I'm not super familiar with this part of lab so not sure to which array the code should be added (or if it should at all).

**Comment by devinivy** [comment]
Good call— here's that patch in lab https://github.com/hapijs/lab/pull/1051

File: package.json [hunk 1]
```diff
@@ -18,16 +18,16 @@
...
+    "@hapi/hoek": "^10.0.0",
+    "@hapi/teamwork": "^6.0.0",
+    "@hapi/validate": "^2.0.0"
   },
   "devDependencies": {
-    "@hapi/code": "8.x.x",
+    "@hapi/code": "^9.0.0",
     "@hapi/eslint-plugin": "*",
-    "@hapi/lab": "24.x.x",
+    "@hapi/lab": "^25.0.0",
```
Comment (devinivy) [inline package.json:28]: Pending https://github.com/hapijs/lab/pull/1051
```suggestion
    "@hapi/lab": "^25.0.1",
```

File: package.json [hunk 2]
```diff
@@ -18,16 +18,16 @@
...
     "@types/node": "^17.0.31",
-    "typescript": "~4.4.4"
+    "typescript": "^4.6.4"
```
Comment (devinivy) [inline package.json:30]:
```suggestion
    "typescript": "~4.6.4"
```

**Review by Marsup — APPROVED** [review]

**Pull request merged**